### DISCUSSION TITLE
Store OAuth provider payload (id_token)

### DISCUSSION THREAD

**Discussion by XaviTorello** [fastapi-users/fastapi-users#657] [Ideas]
Hi there!

We're studying alternatives to be able to interact with the `id_token` data submitted by the provider.

As an example, we want to fill the `User` object with the `name` and `surname` provided by the OAuth `profile` scope.

Which could be the best solution for this?

We're prototyping locally an extension of the `BaseOAuthAccount` model. The idea is to store it when the callback is reached, and use it to fill our `User` inside the `on_after_register` hook directly inspecting it through the `User. oauth_accounts` relation.

This will provide us the possibility to reuse provider's data (based on requested scopes) and use it once the user has been registered (or whenever is needed).

If this approach sounds ok, we're almost ready to submit a PR solving it.

Thoughts?

**Comment by frankie567** [comment]
What I would recommend is to use the `access_token` you retrieve and use it inside `on_after_register` to retrieve the information you need and update your user. Admittedly, it requires to make an extra API call; but you could schedule this operation in a worker if you're worried about response times.

Concerning FastAPI Users, I prefer to keep it simple and focused on user authentication. Different OAuth providers will have different ways of retrieving the user information, so I prefer to leave developers implement their own logic in their projects.
